bundles/org.openhab.binding.opensprinkler/src/main/java/org/openhab/binding/opensprinkler/internal/api/OpenSprinklerGpioApi.java

```diff
@@ -27,8 +28,9 @@
  * that the OpenSprinkler PI device is using.
  *
  * @author Jonathan Giles, Chris Graham - Initial contribution
+ * @author Florian Schmidt - Refactor class visibility
  */
-public class OpenSprinklerGpioApi implements OpenSprinklerApi {
+class OpenSprinklerGpioApi implements OpenSprinklerApi {
```

Comment:
I'm not against that, but why did you change the visibility? The internal packages are not exported from the OSGi-bundle anyway.

Reply:
Yes, not to other bundles, however, this bundle _could_ use the class directly. However, the implementation of the `OpenSprinklerApi` are internal implementation details of the api package. An outside class should not even know that there're different implementations. In my opinion, it makes sense, to enforce this encapsulation on the compiler level as well. That's why I made these classes package private, the factory (which is now even easier to use and change, as it's an OSGi component) and the interface are the only information an outside member of the bundle needs to know and see.

I hope that I explained what I mean in an understandable way :D